.github/workflows/license.yml

```diff
@@ -0,0 +1,38 @@
+name: License
+on:
+  pull_request:
+    branches: [main]
+  schedule:
+    - cron: '0 12 * * *'  # Run daily
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    name: Update BSL change date
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          # We need a commit that modified a file other than the license
+          fetch-depth: 5
+      - name: Check last commit date
+        run: |
+          # The date of the last commit that modified a file other than the license
+          commit_date=$(git log -1 --format=%cd --date=short -- . ':!/LICENSE')
+          [ -z "$commit_date" ] && exit 0
+
+          # The date in the license
+          change_date=$(sed --quiet --regexp-extended 's/^Change Date:\s+([0-9]{4}-[0-9]{2}-[0-9]{2})/\1/p' < LICENSE)
+          [ -z "$change_date" ] && exit 0
+          
+          new_change_date=$(date --date "$commit_date +3 years" --iso-8601)
+
+          if [ "$(date --date "$new_change_date" +%s)" -gt "$(date --date "$change_date" +%s)" ]; then
```

Comment:
wouldn't this spam date update after every change?

Reply:
In my proposal, the cron job would run once per day. The job will update the LICENSE only if there were commits earlier that day. If you feel this is too noisy, we can change the cron schedule to be only once per week or once per month.